src/components/app/pages/questions/index.tsx

```diff
@@ -25,12 +36,132 @@ const Questions = () => {
 		);
 	}, []);
 
+	useEffect(() => {
+		if (user != undefined) {
+			user.questionIndex = userIndex;
+		}
+
+		return () => {
+			if (user != undefined) {
+				writeCurrentProgress(user);
+			}
+		};
+	}, [userIndex]);
+
 	const circleButtonClicked = (question: Question, index: number) => {
 		setUserIndex(index);
 	};
 
+	const loadUser = () => {
+		const userAuth = useAuth().currentUser;
+		if (userAuth != undefined) {
+			User.checkForUserForExam(
+				parseInt(location.state as string),
+				userIndex,
+				userAuth.uid,
+				userAuth.email ?? "",
+			).then((user) => {
+				setUser(user);
+				setUserIndex(user.questionIndex);
+			});
+		}
+	};
+
+	const multipleChoiceQuestionClicked = (
+		question: Question,
+		index: number,
+	) => {
+		const answer = [question.answers[index]];
+		const correct = question.correctAnswers() === answer;
+		const attemptedAnswer = new AttemptedAnswer(
+			question.id,
+			correct,
+			answer.map((x) => {
+				return x.answerID;
+			}),
+			undefined,
+		);
+		user?.addOrUpdateAnswer(attemptedAnswer);
+	};
+
+	const multipleChoiceMultipleCorrectQuestionClicked = (
+		question: Question,
+		index: Array<number | undefined>,
+	) => {
+		const answers = index.flatMap((x) => {
+			return x === undefined ? [] : [question.answers[x]];
+		});
+		let isCorrect = true;
```

Comment:
Answers are not actually being assessed as correct or not. Everything is being set as false currently

Reply:
This code is untested but I think it is correct. My vote is we merge and make sure to test once multiple choice multiple correct questions are in the database.